homeassistant/components/proliphix/climate.py

```diff
@@ -37,6 +37,7 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     host = config.get(CONF_HOST)
 
     pdp = proliphix.PDP(host, username, password)
+    pdp.update()
```

Comment:
This should not be needed since we call `add_entities` with second argument `True` below. That makes sure that the entity `update` method is called during entity addition. The entity `update` method calls `self._pdp.update`.

```suggestion
```

Reply:
Hi,

I tried removing it, but then the error returns:

```
2020-04-22 18:59:15 ERROR (MainThread) [homeassistant.components.climate] Error while setting up proliphix platform for climate
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 178, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/proliphix/climate.py", line 41, in setup_platform
    add_entities([ProliphixThermostat(pdp)], True)
  File "/usr/src/homeassistant/homeassistant/components/proliphix/climate.py", line 50, in __init__
    self._name = self._pdp.name
  File "/usr/local/lib/python3.7/site-packages/proliphix/proliphix.py", line 184, in name
    return self._data['DevName']
KeyError: 'DevName'
```

Maybe the update is called after the name property is retrieved?

Reply:
Ah, I see, in the add_entities call, the ProliphixThermostat constructor is called. That constructor tries to access the self._pdp.name poperty, which has not been set yet.

I can think of several ways to solve this, I am not sure what the preferred one is?
1) Leave the pdp.update() call where I put it initially.
2) Have ProliphixThermostat.__init__ call the update function
3) Have ProliphixThermostat.update set the self._name after the call to self._pdp.update

Reply:
Set name in `update`. In init default name to `None`.

Reply:
Hi All, my apologies if this is not the right spot for a message but I am available to help test.  I have the nt150 thermostat and HA running a working, except for this "devname" error.

Reply:
Hi MartinHjelmare,
Yes, that works as well. The pull request has been updated.